### ISSUE TITLE
Enforce switching to github user before running commands

### ISSUE THREAD

**Issue by uwla** [pwntester/octo.nvim#743]
GH allows switching between user profiles with `gh auto switch`.

Is there a way I can configure `octo.nvim` to always switch to `<user>`
before running any command?

For example, suppose my default github user is `user1` but I have a
project called `project2` which I use `user2` and would like to use
`user2` for this project. If I have configured `octo.nvim` to use `user2`,
then it would switch to this user before running any `octo` command.

The manual solution is to run `gh auth switch` and pick `user2` every
time I open `project2` in neovim.

Another option would be to configure `octo.nvim` such that it always
switch to the configured `user` before calling `gh` behind the scenes.

This would enable us to have a default `user1` and to modify this value
on a per-project basis using other nvim plugins which load project
configuration from files `.vim`, `.vim/`, `init.lua`, etc. That way,
`project2` can modify the default `user1` to `user2` like so:

```lua
-- this is a local file to project2, which modifies default settings
require('octo.nvim').setup({
    default_user = 'user2'
})
```

**Comment by wd60622** [comment]
Yeah, I think that it'd be a good idea to at least not blindly do commands without knowing which username it is from.

It might be good to expose the current user somehow so other plugins can use it. For instance, displaying it in the statusline.

Exposing `gh auth switch` would be a good idea too.

What are your thoughts? Would like you like to open a PR for this?

**Comment by uwla** [comment]
I have made a workaround:

```lua
-- OCTO wrapper command
vim.api.nvim_create_user_command('GH', function(args)
    local gh_user = vim.g.gh_auth_user
    if gh_user ~= nil then
        vim.fn.system("gh auth switch --user " .. gh_user)
    end
    require('octo')
    vim.cmd('Octo ' .. table.concat(args.fargs, ' '))
end, { nargs = '*' })
```

Instead of calling `Octo`, I call the `GH` command which switches the user
behind the scenes. That way, I can set `vim.g.gh_auth_user` on a per-project
basis and have a default one.

The main problem is that I do not get completion in the cmdline.

> What are your thoughts? Would like you like to open a PR for this?

I could go for that even though I am not familiar with Lua (I only use it for neovim)

I'll take a look at the codebase this week or next.
